### PR TITLE
Make component searching fuzzy

### PR DESCRIPTION
**Pull request by bogdan-stefan-hs**
## Need

``` gherkin
As a user
I want a smart searching algorithm
So that I can find the desired component with less keystrokes.
```
## Deliverables
- The component search bar uses fuzzy searching, approximately matching items
## Solution
- Use [fuzzaldrin-plus](https://github.com/jeancroy/fuzzaldrin-plus) for the fuzzy searching algorithm.

### CODE REVIEW

**Comment by bogdan-stefan-hs** [comment]
@flavius-tirnacop-hs @valentin-radulescu-hs Please review code.

fuzzy

File: src/components/component-playground.jsx [hunk 1]
```diff
@@ -554,48 +558,44 @@ module.exports = React.createClass({
...
-    }.bind(this)),
+  _getFilteredComponents() {
+    if (this.state.searchText.length < 2) {
+      return this.props.components;
+    }
+
+    // Get components which match search text by one or more fixture names, only
+    // keeping those fixtures. Also keep the selected fixture.
+    return _.pick(_.mapValues(this.props.components,
+        function(details, componentName) {
```
Comment (valentin-radulescu-hs) [inline src/components/component-playground.jsx:593]: You can reduce indentation here and remove `.bind(this)` by using fat arrow 

``` js
const {components} = this.props;


return _.pick(_.mapValues(components, (details, componentName) => { /*...*/ }));

```

File: src/components/component-playground.jsx [hunk 2]
```diff
@@ -554,48 +558,44 @@ module.exports = React.createClass({
...
+          var maxScore = 0;
+          return _.assign({}, details, {
+          // Only select fixtures matching search text and rebuild the object.
+            fixtures: _.pick(details.fixtures,
+                function(fixtureProps, fixtureName) {
```
Comment (valentin-radulescu-hs) [inline src/components/component-playground.jsx]: Same as [this](https://github.com/uberVU/react-component-playground/pull/8/files#r74903210)

File: src/components/component-playground.jsx [hunk 3]
```diff
@@ -554,48 +558,44 @@ module.exports = React.createClass({
...
-      // Only select fixtures matching search text and rebuild the object.
-        fixtures: _.pick(details.fixtures,
-            function(fixtureProps, fixtureName) {
-              return fixtureName.indexOf(this.state.searchText) !== -1;
-            }.bind(this)
-        )
-      });
-    }.bind(this)),
+  _getFilteredComponents() {
+    if (this.state.searchText.length < 2) {
```
Comment (valentin-radulescu-hs) [inline src/components/component-playground.jsx:586]: We're going to need a test for this

File: src/components/component-playground.jsx (hunk 1 above)
Comment (bogdan-stefan-hs) [inline src/components/component-playground.jsx:593]: The problem here is that, as it was suggested in a previous PR, `component-playground.jsx` does not have any ES6 features, so the style would get out of sync. We can create a new issue for adding some ES6 features, if everybody agrees.

Comment (valentin-radulescu-hs) [inline src/components/component-playground.jsx:593]: Gotcha

**Comment by valentin-radulescu-hs** [comment]
Finished review.

**Comment by bogdan-stefan-hs** [comment]
@valentin-radulescu-hs I've added the test for the small search text length.
cc @flavius-tirnacop-hs

**Comment by flavius-tirnacop-hs** [comment]
LGTM

**Comment by valentin-radulescu-hs** [comment]
LGTM

**Pull request merged**